src/robot/libraries/BuiltIn.py

```diff
@@ -876,31 +970,51 @@ def get_count(self, item1, item2):
         self.log('Item found from the first item %d time%s' % (count, s(count)))
         return count
 
-    def should_not_match(self, string, pattern, msg=None, values=True):
+    def should_not_match(self, string, pattern, msg=None, values=True, ignore_case=False):
         """Fails if the given ``string`` matches the given ``pattern``.
 
         Pattern matching is similar as matching files in a shell, and it is
         always case-sensitive. In the pattern ``*`` matches to anything and
         ``?`` matches to any single character.
 
+        ``ignore_case`` is False by default.  If True, it indicates that
+        ``first`` and ``second`` should be compared case-insensitively,
+        provided that ``first`` and ``second`` are string types.  See `Boolean
+        arguments` section for more details.  (New in RF 3.0.1)
+
         See `Should Be Equal` for an explanation on how to override the default
         error message with ``msg`` and ``values``.
         """
-        if self._matches(string, pattern):
+        prv_string = string
+        prv_pattern = pattern
+        if is_truthy(ignore_case) and is_string(string) and is_string(pattern):
+            prv_string = string.lower()
+            prv_pattern = pattern.lower()
+        if self._matches(prv_string, prv_pattern):
```

Comment:
No need to explicitly check are `string` and `pattern` strings or not. Simply using

```python
if is_truthy(ignore_case):
    string = string.lower()
    pattern = pattern.lower()
```

ought to be enough.